In [1]:
import numpy as np
import pandas as pd
import pickle

from markovchain import MarkovChain

In [2]:
with open("input/seq_events.txt", "rb") as fp:
    seq_events = pickle.load(fp)

In [3]:
lengths = []
for i in seq_events:
    lengths.append(len(i))
    
print ('Mean Length: '+str(np.mean(lengths)))
print ('Minimum Length: '+str(np.min(lengths)))
print ('Maximum Length: '+str(np.max(lengths)))

Mean Length: 47.133057151766224
Minimum Length: 1
Maximum Length: 1466


<table><td>Seq Length 1 -> Count 38
Seq Length 2 -> Count 5237
Seq Length 3 -> Count 1481
Seq Length 4 -> Count 4959
Seq Length 5 -> Count 1903
Seq Length 6 -> Count 549
Seq Length 7 -> Count 114
Seq Length 8 -> Count 49
Seq Length 9 -> Count 49
Seq Length 10 -> Count 27
Seq Length 11 -> Count 65
Seq Length 12 -> Count 12
Seq Length 13 -> Count 65
Seq Length 14 -> Count 53
Seq Length 15 -> Count 326
Seq Length 16 -> Count 112
Seq Length 17 -> Count 19
Seq Length 18 -> Count 35
Seq Length 19 -> Count 306
Seq Length 20 -> Count 15
Seq Length 21 -> Count 13
Seq Length 22 -> Count 6
Seq Length 23 -> Count 482
Seq Length 24 -> Count 7
Seq Length 25 -> Count 4
Seq Length 26 -> Count 119
Seq Length 27 -> Count 5
Seq Length 28 -> Count 52
Seq Length 29 -> Count 17
Seq Length 30 -> Count 19
Seq Length 31 -> Count 637
Seq Length 32 -> Count 5
Seq Length 34 -> Count 3
Seq Length 35 -> Count 1
Seq Length 36 -> Count 3
Seq Length 38 -> Count 2
Seq Length 39 -> Count 2
Seq Length 40 -> Count 3
Seq Length 41 -> Count 1
Seq Length 42 -> Count 3
Seq Length 43 -> Count 2
Seq Length 44 -> Count 11
Seq Length 45 -> Count 1
Seq Length 46 -> Count 3
Seq Length 48 -> Count 2
Seq Length 50 -> Count 2
Seq Length 51 -> Count 3
Seq Length 52 -> Count 1
Seq Length 53 -> Count 2
Seq Length 54 -> Count 2
Seq Length 55 -> Count 3
Seq Length 56 -> Count 2
Seq Length 58 -> Count 3
Seq Length 59 -> Count 1
Seq Length 60 -> Count 3
Seq Length 62 -> Count 2
Seq Length 63 -> Count 6
Seq Length 64 -> Count 1
Seq Length 66 -> Count 3
Seq Length 68 -> Count 1
Seq Length 69 -> Count 188
Seq Length 70 -> Count 1
Seq Length 71 -> Count 659
Seq Length 75 -> Count 1
Seq Length 76 -> Count 22
Seq Length 79 -> Count 1
Seq Length 80 -> Count 37
Seq Length 81 -> Count 2
Seq Length 83 -> Count 1
Seq Length 85 -> Count 1
Seq Length 88 -> Count 1
Seq Length 90 -> Count 1
Seq Length 92 -> Count 2
Seq Length 94 -> Count 1
Seq Length 95 -> Count 1
Seq Length 96 -> Count 1
Seq Length 97 -> Count 2<td></
Seq Length 98 -> Count 2
Seq Length 99 -> Count 1
Seq Length 100 -> Count 1
Seq Length 101 -> Count 2
Seq Length 106 -> Count 9
Seq Length 111 -> Count 1
Seq Length 112 -> Count 2
Seq Length 114 -> Count 1
Seq Length 121 -> Count 1
Seq Length 122 -> Count 9
Seq Length 123 -> Count 1
Seq Length 124 -> Count 2
Seq Length 126 -> Count 2
Seq Length 127 -> Count 31
Seq Length 128 -> Count 1
Seq Length 129 -> Count 625
Seq Length 132 -> Count 1
Seq Length 133 -> Count 8
Seq Length 134 -> Count 1
Seq Length 135 -> Count 1
Seq Length 137 -> Count 1
Seq Length 143 -> Count 2
Seq Length 145 -> Count 1
Seq Length 152 -> Count 1
Seq Length 157 -> Count 1
Seq Length 161 -> Count 1
Seq Length 166 -> Count 2
Seq Length 171 -> Count 1
Seq Length 175 -> Count 2
Seq Length 177 -> Count 3
Seq Length 179 -> Count 1
Seq Length 183 -> Count 1
Seq Length 186 -> Count 1
Seq Length 188 -> Count 18
Seq Length 191 -> Count 1
Seq Length 192 -> Count 8
Seq Length 193 -> Count 2
Seq Length 194 -> Count 443
Seq Length 197 -> Count 418
Seq Length 199 -> Count 3032
Seq Length 205 -> Count 1
Seq Length 206 -> Count 1
Seq Length 209 -> Count 1
Seq Length 210 -> Count 1
Seq Length 216 -> Count 1
Seq Length 236 -> Count 1
Seq Length 239 -> Count 1
Seq Length 243 -> Count 1
Seq Length 252 -> Count 1
Seq Length 257 -> Count 1
Seq Length 264 -> Count 1
Seq Length 265 -> Count 1
Seq Length 266 -> Count 1
Seq Length 271 -> Count 1
Seq Length 293 -> Count 1
Seq Length 304 -> Count 1
Seq Length 309 -> Count 1
Seq Length 316 -> Count 1
Seq Length 320 -> Count 1
Seq Length 348 -> Count 1
Seq Length 372 -> Count 1
Seq Length 403 -> Count 1
Seq Length 425 -> Count 1
Seq Length 427 -> Count 1
Seq Length 509 -> Count 1
Seq Length 529 -> Count 1
Seq Length 545 -> Count 1
Seq Length 713 -> Count 1
Seq Length 1054 -> Count 1
Seq Length 1090 -> Count 1
Seq Length 1326 -> Count 1
Seq Length 1408 -> Count 1
Seq Length 1466 -> Count 1</td></table>

In [4]:
mod = MarkovChain(states=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14])

In [5]:
mod.train(seq_events)

In [6]:
mod.get_probability([12, 1, 10])

0.25082964115440604

In [7]:
mod.transition[3][4]

0.9199865877900989